llvm/lib/ProfileData/SampleProf.cpp

```diff
@@ -401,12 +401,26 @@ std::error_code ProfileSymbolList::read(const uint8_t *Data,
   const char *ListStart = reinterpret_cast<const char *>(Data);
   uint64_t Size = 0;
   uint64_t StrNum = 0;
+  uint64_t ExpectedCount = 0;
+
+  // Scan forward to see how many elements we expect.
+  while (Offset < ListSize) {
```

Comment:
It would be cleaner to have exactly the same condition

```
while (Size < ListSize && StrNum < ProfileSymbolListCutOff) {
  ExpectedCount++;
 Size++
}
```

As is, I am not sure understand why 
Only  `ListStart[Offset] == '\0'` needs a reserve?